### ISSUE TITLE
fmt fails when passed a file or subdirectory

### ISSUE THREAD

**Issue by rosensama** [golangci/golangci-lint#6417]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/docs/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

Official binary

### Description of the problem

In past releases (last checked was 2.8.0) I have been able to run `golangci-lint fmt <path/to/file.go>` or `golangci-lint fmt <path/to/directory/>`.   These both fail today.

```console
$ golangci-lint fmt database/...
Error: failed to process files: openat database/backgroundpoller.go: no such file or directory
The command is terminated due to an error: failed to process files: openat database/backgroundpoller.go: no such file or directory

$ golangci-lint fmt database/backgroundpoller.go
Error: failed to process files: open database/backgroundpoller.go: not a directory
The command is terminated due to an error: failed to process files: open database/backgroundpoller.go: not a directory
```

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 2.11.1 built with go1.26.1 from 89a46a24 on 2026-03-06T14:04:16Z
```

</details>

### Configuration

<details>

```yaml
version: "2"

linters:
  default: all # Enable all linters by default
  disable:
    # Overly strict style linters
    - exhaustruct # Requires all struct fields (too strict)
    - varnamelen # Variable name length (opinionated)
    - wsl # Whitespace linting (opinionated)
    - wsl_v5
    - nlreturn # Blank line before return (opinionated)
    - gochecknoglobals # No globals (too strict for some use cases)
    - inamedparam # Named parameters (too strict)
    - tagliatelle # Struct tag format (opinionated)
    - depguard # Blocks all imports by default, needs complex allow-list config (too strict)
    - godox # Flags all TODO comments (too noisy)
    - err113 # Requires sentinel errors (opinionated)
    - godot # Requires all comments longer than X to end in period (too strict)
    - noinlineerr # this is contrary to Google style guide https://google.github.io/styleguide/go/guide#concision gocritic might have something that recommends inlining in some cases (just wrong)

    # Context-specific (enable when ready for deeper analysis)
    - funlen # Function length limits
    - gocognit # Cognitive complexity
    - gocyclo # Cyclomatic complexity
    - cyclop # Complexity
    - maintidx # Maintainability index
    #    - nestif # Nested if depth
    - dupl # Duplicate code (too strict, identifies patterns that could probably be DRY'd up, but aren't exact dupes)

    # Test-specific (may conflict with Ginkgo/Gomega patterns)
    - testpackage # Separate _test packages (conflicts with Ginkgo)
    - paralleltest # Require t.Parallel() (Ginkgo handles this differently)

  settings:
    lll:
      line-length: 120

    errcheck:
      check-type-assertions: true

    govet:
      enable-all: true
      disable:
        - fieldalignment # if memory footprint and gc ever matter come back to this on oft-used structs

    staticcheck:
      checks: ["all"]

    gosec:
      excludes:
        - G404 # Weak random number generator - OK for non-crypto use

    revive:
      rules:
        # Disable some opinionated rules
        - name: add-constant
          disabled: true
        - name: cognitive-complexity
          disabled: true
        - name: function-length
          disabled: true
        - name: max-public-structs
          disabled: true

    gocritic:
      enabled-tags:
        - diagnostic
        - style
        - performance
      disabled-checks:
        - hugeParam # Often triggers on generated code

    gomoddirectives:
      replace-allow-list:
        - github.com/simukti/sqldb-logger
      tool-forbidden: true
      toolchain-forbidden: true

    recvcheck:
      # I think the implementation makes a bad assumption that if Marshal and UnMarshal are both implemented then
      # everything but Marshal will be a pointer receiver.  In this code base Secret is all val type except Unmarshals
      disable-builtin: false
      exclusions:
        - "Secret.UnmarshalJSON"
        - "Secret.UnmarshalText"

  exclusions:
    generated: strict
    presets:
      - comments
      - common-false-positives
      - legacy
      # std-error-handling intentionally omitted: we want to catch bare Close() calls
      # Use log.DeferredClose() or `_ = foo.Close()` to explicitly handle/discard errors
    paths:
      - third_party$
      - builtin$
      - examples$
    rules:
      # Allow dot imports for Ginkgo/Gomega in test files (standard BDD practice)
      # Test files can also have longer functions
      - path: _test\.go
        linters:
          - staticcheck
          - revive
          - stylecheck
          - funlen
          - gocognit
          - gocyclo
        text: "dot imports|ST1001"

      - path: fixtures/
        linters:
          - mnd

      - path: testutil/
        linters:
          - staticcheck
          - revive
          - stylecheck
        text: "dot imports|ST1001"

      # Generated code exclusions
      - path: _string\.go$
        linters:
          - all

      - path: mock_.*\.go$
        linters:
          - all

      # Allow globals in main.go (common pattern)
      - path: main\.go$
        linters:
          - gochecknoglobals

      # Allow fmt.PrintX
      - path: tools/playground/
        linters:
          - forbidigo

      # Allow fmt.PrintX
      - path: tools/sqlx/
        linters:
          - forbidigo

formatters:
  enable:
    - goimports
  settings:
    goimports:
      local-prefixes:
        - mendoza-server

run:
  timeout: 5m
  tests: true

issues:
  max-issues-per-linter: 0
  max-same-issues: 0
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version && go env
go version go1.26.1 darwin/arm64
AR='ar'
CC='clang'
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_ENABLED='1'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
CXX='clang++'
GCCGO='gccgo'
GO111MODULE=''
GOARCH='arm64'
GOARM64='v8.0'
GOAUTH='netrc'
GOBIN='/Users/jobu/.asdf/installs/golang/1.26.1/bin'
GOCACHE='/Users/jobu/Library/Caches/go-build'
GOCACHEPROG=''
GODEBUG=''
GOENV='/Users/jobu/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFIPS140='off'
GOFLAGS=''
GOGCCFLAGS='-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/nm/nfmnz8cs11n7xd17qr854zn40000gn/T/go-build3106195257=/tmp/go-build -gno-record-gcc-switches -fno-common'
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMOD='/Users/jobu/GolandProjects/mendoza-server/go.mod'
GOMODCACHE='/Users/jobu/.asdf/installs/golang/1.26.1/packages/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='darwin'
GOPATH='/Users/jobu/.asdf/installs/golang/1.26.1/packages'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/Users/jobu/.asdf/installs/golang/1.26.1/go'
GOSUMDB='sum.golang.org'
GOTELEMETRY='local'
GOTELEMETRYDIR='/Users/jobu/Library/Application Support/go/telemetry'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/Users/jobu/.asdf/installs/golang/1.26.1/go/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.26.1'
GOWORK=''
PKG_CONFIG='pkg-config'
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint fmt -v
golangci-lint fmt -v
INFO golangci-lint has version 2.11.1 built with go1.26.1 from 89a46a24 on 2026-03-06T14:04:16Z 
INFO [config_reader] Config search paths: [./ /Users/jobu/GolandProjects/mendoza-server /Users/jobu/GolandProjects /Users/jobu /Users /] 
INFO [config_reader] Used config file .golangci.yml 
INFO [config_reader] Module name "mendoza-server" 
INFO Formatting Go files...                       
INFO format: webapi/rest/middleware/resbodydump.go 

$ golangci-lint fmt -v database
INFO golangci-lint has version 2.11.1 built with go1.26.1 from 89a46a24 on 2026-03-06T14:04:16Z 
INFO [config_reader] Config search paths: [./ /Users/jobu/GolandProjects/mendoza-server/database /Users/jobu/GolandProjects/mendoza-server /Users/jobu/GolandProjects /Users/jobu /Users /] 
INFO [config_reader] Used config file .golangci.yml 
INFO [config_reader] Module name "mendoza-server" 
INFO Formatting Go files...                       
Error: failed to process files: openat database/backgroundpoller.go: no such file or directory
The command is terminated due to an error: failed to process files: openat database/backgroundpoller.go: no such file or directory
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a [sponsor/backer](https://donate.golangci.org) of this project.

**Comment by rosensama** [comment]
It appears this was introduced in 2.11.0

**Comment by ldez** [comment]
I found the problem, I will fix that quickly.
